### PR TITLE
Add task to build and push RPM to a COPR repository

### PR DESCRIPTION
**Pull request by chmouel**
# Changes

Add a task to build and push rpm to a copr repo, everything should be detailed in the README

This supports, manpages and tekton completion (if you install the `bash-completions` package for bash or use zsh)

Todo next: 

* Make sure this all works properly and plug into into the `release-pipeline.yaml`
* Make the Provides/BuildRequires generated automatically
* Make it more generic for task catalog and handle specifying rebuilding a specific version that is not master,

# Release-Note 

```
 COPR repo for tektoncd-cli is now available for Fedora 30 31, Centos8 or RHEL8, to enable it you can simply do :

$ dnf copr enable chmouel/tektoncd-cli

and after this being enable the tektoncd-cli package will be available for install : 

$ dnf install tektoncd-cli
```

Closes: #110 

/cc @vdemeester @khrm

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,14 @@ Download the latest binary executable for your operating system:
   sudo tar xvzf tkn_0.3.1_Linux_arm64.tar.gz -C /usr/local/bin/ tkn
   ```
 
+  You can also use [@chmouel](https://github.com/chmouel)'s unofficial
+  copr package repository for the time being.
+
+  ```shell
+  dnf copr enable chmouel/tektoncd-cli
```
Comment (khrm) [inline README.md]: Nit: Maybe a mention somewhere that this is for fedora?

File: tekton/rpmbuild/tekton.spec [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+install -d %{buildroot}%{_datadir}/bash-completion/completions
+./tkn completion bash > %{buildroot}%{_datadir}/bash-completion/completions/_tkn
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%files
+%defattr(-,root,root,-)
+%doc *.md LICENSE
```
Comment (khrm) [inline tekton/rpmbuild/tekton.spec]: Add man files also to appropriate location.

Comment (chmouel) [reply tekton/rpmbuild/tekton.spec]: I don't have access to manpage from a binary build :(

File: tekton/rpmbuild/tekton.spec [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+BuildRequires: git
+
+%global import_path github.com/tektoncd/cli
+
+Source0:        https://%{repo}/archive/v%{version}.tar.gz
+
+%description
+The Tekton Pipelines cli project provides a CLI for interacting with Tekton !
+
+# vendored libraries (Source0)
```
Comment (vdemeester) [inline tekton/rpmbuild/tekton.spec]: Are those generated with a tool ?

Comment (chmouel) [reply tekton/rpmbuild/tekton.spec]: @khrm   was saying so

what about we merge (when i finish to iterate on it) this and let khuram updating it with automatic generation of the deps if needed,

File: tekton/rpmbuild/tekton.spec (hunk 2 above)
Comment (chmouel) [reply tekton/rpmbuild/tekton.spec]: moved to your spec file and add the manpage thee,

File: tekton/rpmbuild/tekton.spec (hunk 3 above)
Comment (vdemeester) [reply tekton/rpmbuild/tekton.spec]: @chmouel yeah works for me :wink:

**Comment by chmouel** [comment]
/hold cancel 

it's working now! https://copr.fedorainfracloud.org/coprs/chmouel/tektoncd-cli/build/1034136/

**Comment by chmouel** [comment]
CLA don't allow me to add a ```Co-authored-by: Khurram Baig <kbaig@redhat.com>``` 🙀

File: README.md (hunk 1 above)
Comment (chmouel) [reply README.md]: Thanks!

**Pull request merged**